### PR TITLE
Workflow Executions: Update Model Tests

### PR DESCRIPTION
**Pull request by ChrisHuynh333**
## What does this PR do and why?
This PR updates testing the workflow execution model to increase coverage

## Screenshots or screen recordings
_Screenshots are required for UI changes, and strongly recommended for all other pull requests._

## How to set up and validate locally
1. Ensure all tests pass locally

## PR acceptance checklist
This checklist encourages us to confirm any changes have been analyzed to reduce risks in quality, performance, reliability, security, and maintainability.

- [x] I have evaluated the [PR acceptance checklist](https://phac-nml.github.io/irida-next/docs/development/development_processes/code_review#acceptance-checklist) for this PR.

### CODE REVIEW

File: test/models/workflow_execution_test.rb [hunk 1]
```diff
@@ -94,4 +94,173 @@ def setup
...
+    assert @workflow_execution_valid.sent_to_ga4gh?
+
+    @workflow_execution_valid.state = :prepared
+    assert_not @workflow_execution_valid.sent_to_ga4gh?
+
+    @workflow_execution_valid.state = :initial
+    assert_not @workflow_execution_valid.sent_to_ga4gh?
+  end
+
+  test 'send_email' do
```
Comment (ericenns) [inline test/models/workflow_execution_test.rb:185]: **nitpick (blocking):** If I run this test case it reports no assertions have been made. If I refactor lines 190 - 194 with the following I actually see assertions.

```ruby
assert_enqueued_emails 1 do
      workflow_execution_completed.send_email
      assert_enqueued_email_with PipelineMailer, :complete_email, args: [workflow_execution_completed]
    end
```

Note: this example will fail as the fixture you are using for this test does not have `email_notification: true`.

Can you refactor all of these case in your test. The issue is that `workflow_execution_completed.send_email do` is not valid syntax. That would only work if the method accepted a block but then again you wouldn't pass test assertions to a block.

**Review by ericenns — CHANGES_REQUESTED** [review]
This is looking better except that the tests for `send_email` are not working as you intended.

File: test/models/workflow_execution_test.rb (hunk 1 above)
Comment (ChrisHuynh333) [reply test/models/workflow_execution_test.rb:185]: Thanks for catching this. Fixed in 9eaca37fe764f2a38bec728a104ef44e632e25b5

Comment (ChrisHuynh333) [reply test/models/workflow_execution_test.rb:185]: I realized it doesn't make much sense to have `assert_enqueued_email_with PipelineMailer, :complete_email, args: [workflow_execution_example]` in blocks that have 0 enqueued emails. Removed in 00276fea505da70d2136e5c087deb906a3f2f6bd

**Review by ericenns — APPROVED** [review]
Looks good to me!

**Pull request merged**